rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/reporting/rules/CreateDependencyGraphReportRuleProvider.java

```diff
@@ -0,0 +1,96 @@
+package org.jboss.windup.rules.apps.java.reporting.rules;
+
+import org.jboss.windup.config.AbstractRuleProvider;
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.config.loader.RuleLoaderContext;
+import org.jboss.windup.config.metadata.RuleMetadata;
+import org.jboss.windup.config.operation.GraphOperation;
+import org.jboss.windup.config.phase.ReportGenerationPhase;
+import org.jboss.windup.graph.GraphContext;
+import org.jboss.windup.graph.model.ProjectModel;
+import org.jboss.windup.graph.model.resource.FileModel;
+import org.jboss.windup.graph.service.WindupConfigurationService;
+import org.jboss.windup.reporting.model.ApplicationReportModel;
+import org.jboss.windup.reporting.model.TemplateType;
+import org.jboss.windup.reporting.service.ApplicationReportService;
+import org.jboss.windup.reporting.service.ReportService;
+import org.jboss.windup.rules.apps.java.condition.SourceMode;
+import org.ocpsoft.rewrite.config.Configuration;
+import org.ocpsoft.rewrite.config.ConfigurationBuilder;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+import com.google.common.collect.Iterables;
+
+import java.util.List;
+
+/**
+ * Generates the dependency graph reports for the applications analyzed
+ */
+@RuleMetadata(phase = ReportGenerationPhase.class)
+public class CreateDependencyGraphReportRuleProvider extends AbstractRuleProvider
+{
+   public static final String REPORT_NAME = "Dependencies Graph";
+   public static final String REPORT_DESCRIPTION = "This graph shows the dependencies embedded within the analyzed application.";
+   public static final String TEMPLATE = "/reports/templates/dependency_graph.ftl";
+   private static final String REPORT_BASEFILENAME = "dependency_graph_report";
+
+   @Override
+   public Configuration getConfiguration(RuleLoaderContext ruleLoaderContext)
+   {
+      return ConfigurationBuilder.begin()
+               .addRule()
+               .when(SourceMode.isDisabled())
+               .perform(new CreateDependencyGraphReportOperation());
+   }
+
+   private class CreateDependencyGraphReportOperation extends GraphOperation
+   {
+
+      @Override
+      public void perform(GraphRewrite event, EvaluationContext context)
+      {
+         List<FileModel> inputPaths = WindupConfigurationService.getConfigurationModel(event.getGraphContext())
+                  .getInputPaths();
+         if (Iterables.size(inputPaths) > 1)
```

Comment:
Why this Iterables.size instead of a simple inputPaths.size() ? Both choices are not "null protected"

Reply:
+1 - I suspect copy'pasta as some of these used to be Iterables instead of Lists.

Reply:
Good catch: i'm going to change it.
And yes, as Jess highlighted, some of these are defined as `Iterable`s